nemo/collections/nlp/data/data_utils/data_preprocessing.py

```diff
@@ -461,9 +504,36 @@ def get_freq_weights(label_freq):
     for lf in label_freq.values():
         total_size += lf
     weighted_slots = {label: (total_size / (len(label_freq) * freq)) for label, freq in label_freq.items()}
+
     return weighted_slots
 
 
+def get_freq_weights_bce_with_logits_loss(label_freq):
+    """
+    Calculate positive class weights in regards to BCEWithLogitsLoss
+    https://pytorch.org/docs/1.9.1/generated/torch.nn.BCEWithLogitsLoss.html
+
+    Args: 
+        label_freq: dictionary of tuples where keys represents class id, and tuple represents counts of positive and negative classes,
+                    positive classes are at index 1 and negative at index 0
+    Returns:
+        weights: dictionary of labels with their weights
+    """
+    weights = {}
+
+    for label_id, class_values in label_freq.items():
+        positive_class = class_values[1]
+        negative_class = class_values[0]
+
+        if positive_class == 0:
+            weights[label_id] = 0
+
+        else:
+            weights[label_id] = float(negative_class) / float(positive_class)
```

Comment:
I think this should be negative/total   (i.e. negative/(negative + positive)

Reply:
I think from https://pytorch.org/docs/1.9.1/generated/torch.nn.BCEWithLogitsLoss.html it says that the pos_weight for a class should be negative / positive example (The paragraph right above "Examples:"

Reply:
Ah yes, you're right, of course.